wgengine/magicsock/endpoint.go

```diff
@@ -354,6 +359,44 @@ func (de *endpoint) addrForWireGuardSendLocked(now mono.Time) (udpAddr netip.Add
 	return udpAddr, needPing
 }
 
+// addrForPingSizeLocked returns the address(es) that should be used for sending
+// the next ping. It will only return addrs with a large enough path MTU to
+// permit a ping payload of size bytes to be delivered (DERP is always one such
+// addr as it is a TCP connection). If it returns a zero-value udpAddr, then we
+// should continue probing the MTU of all paths to this endpoint. Zero, one, or
+// both of the returned UDP address and DERP address may be non-zero.
+//
+// de.mu must be held.
+func (de *endpoint) addrForPingSizeLocked(now mono.Time, size int) (udpAddr, derpAddr netip.AddrPort) {
+	if size == 0 {
+		udpAddr, derpAddr, _ = de.addrForSendLocked(now)
+		return
+	}
+
```

Comment:
I can see that `de.isWireguardOnly` has special treatment for wireguard-only peers. Should we handle them in some special way as well when `size` is not zero?

Reply:
Good question. I *think* that we can only send a CLI disco ping to a Tailscale peer because Wireguard-only peers don't understand disco. And the call of addrForSendLocked() that this replaced ignored the startWGPing return value. @raggi and @catzkorn - is it possible to get to this point in the code with a Wireguard-only peer?